cookbooks/bach_krb5/recipes/upload_keytabs.rb

```diff
@@ -0,0 +1,22 @@
+require 'base64'
+# Upload keytabs to chef-server
+get_all_nodes().each do |h|
+  node[:bcpc][:hadoop][:kerberos][:data].each do |srvc, srvdat|
+    # Set host based on configuration 
+    config_host=srvdat['princhost'] == "_HOST" ? float_host(h[:hostname]) : srvdat['princhost'].split('.')[0]
+    keytab_host=srvdat['princhost'] == "_HOST" ? float_host(h[:fqdn]) : srvdat['princhost']
+
+    # Delete existing configuration item (if requested)
+    delete_config("#{config_host}-#{srvdat['principal']}") if node[:bcpc][:hadoop][:kerberos][:keytab][:recreate] == true
```

Comment:
This does depend on `BCPC`. Please see [here](https://github.com/bloomberg/chef-bach/blob/kerberos/cookbooks/bach_krb5/metadata.rb#L9-L11)